tests/utils/test_ensemble.py

```diff
@@ -101,6 +103,26 @@ def create_graphSAGE_model(graph, link_prediction=False):
     return base_model, keras_model, generator, train_gen
 
 
+def create_Unsupervised_graphSAGE_model(graph):
+    generator = GraphSAGELinkGenerator(graph, batch_size=2, num_samples=[2, 2])
+    unsupervisedSamples = UnsupervisedSampler(
+        graph, nodes=graph.nodes(), length=3, number_of_walks=2
+    )
+    train_gen = generator.flow(unsupervisedSamples)
+
+    base_model = GraphSAGE(
+        layer_sizes=[8, 8], generator=train_gen, bias=True, dropout=0.5
```

Comment:
```suggestion
        layer_sizes=[8, 8], generator=generator, bias=True, dropout=0.5
```
I think the modern interface takes a generator now.

Reply:
Using the generator as-is throws an attribute error  'GraphSAGELinkGenerator' object has no attribute 'generator'". I think this may be caused by the `UnsupervisedSampler` function, which makes use of the function `onDemandLinkSequence` which doesn't have a generator attribute. Works like a charm with the older interface though as used in this PR.

Reply:
Hmm, we might need to look at this further.

Reply:
Out of curiosity, where have you branched this from? The docs and behaviour for `GraphSAGE` in `develop` seem different from your branch, especially in regards to generators.

Reply:
I branched from - v0.8.2 tag

Reply:
Is this a hot-fix? Otherwise shouldn't it branch from `develop`? @youph , can you enlighten me here?

Reply:
Righto I branched from the master branch - which was up to date with the develop branch as far as I could tell

Reply:
All good @geoffj-d61, will re-open and re-branch from develop and resend PR

Reply:
Actually on second thought, it is a hotfix, as currently you cannot use unsupervised ensembling on the current v08.3 tag

Reply:
"it is a hotfix, as currently you cannot use unsupervised ensembling on the current v08.3 tag" - do we try to use unsupervised ensembling there, or suggest that it can be done? If not, then this is more like an enhancement that adds functionality, rather than a hotfix, imo.

Reply:
I hackily made this work on a notebook by using the changes in the `ensemble.py` module and thought someone else out there with the current version would run into the same problem, hence the PR. 

I figured it may have been mistakenly left out from the ensemble class. It's a rather minor fix, as you can currently use it to perform training and inference on unsupervised graphsage.

Happy to delete branch/PR and resubmit it as a undone task.

Reply:
I think if you rebranch it from develop, and then you can use the `GraphSAGE(generator=generator)` modification that I suggested.

As @youph  said, this sounds more like an enhancement than an urgent hot-fix.

Reply:
No problemo